src/components/donations/grid/Grid.tsx

```diff
@@ -24,10 +25,12 @@ interface PersonCellProps {
 export default observer(function Grid() {
   const [pageSize, setPageSize] = useState(5)
   const { t } = useTranslation()
-
-  const { data }: UseQueryResult<DonationResponse[]> = useDonationsList()
+  const router = useRouter()
   const { isDetailsOpen } = ModalStore
-
+  const id = router.query.id
+  const { data }: UseQueryResult<DonationResponse[]> = id
+    ? useCampaingDonationsList(id as string)
+    : useDonationsList()
```

Comment:
the id here is the campaignId, so please rename to campaignId for the code to be more readable